src/main/java/uk/gov/hmcts/reform/cpo/auditlog/LogAudit.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.hmcts.reform.cpo.auditlog;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Use this annotation on the endpoint method to create the audit log entry and send to stdout.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.METHOD })
+public @interface LogAudit {
+
+    AuditOperationType operationType();
+
+    String cpoId() default "";
+
+    String cpoIds() default "";
+
+    String caseId() default "";
+
```

Comment:
As a compromise I will consolidate the two sets of ID data fields in the AuditContext so they match the final data values in AuditEntry.  So the processing of these fields will move to AuditAspect from AuditService. This already feels cleaner.

I will however keep the separate LogAudit ID values so the mappings can still be specified as a pointer to either a string or a list value, as otherwise all ID fields would need to be converted to a list or hammered into a CSV string (after apply the max list size filter).  I think this leaves the LogAudit values as a useful shortcut when specifying the mappings and keeps the log formatting rules together rather than in the controller.

Reply:
sure make sense.

Reply:
Resolved in 8fcbda9faaefab2c281ae0d6bc0cee8f12cbcac2.